sec1_intro_and_setup/test_app/app/assets/config/manifest.js

```diff
@@ -0,0 +1,2 @@
+//= link_tree ../images
```

Comment:
`manifest.js` is a JSON file that tells the browser about your Progressive Web App and how it should behave when installed on the user's desktop or mobile device. A typical manifest file includes the app name, the icons the app should use, and the URL that should be opened when the app is launched. When compiling assets with Sprockets, Sprockets needs to decide which top-level targets to compile. The `manifest.js` file is meant to specify which files to use as a top-level target using sprockets methods `link`, `link_directory`, and `link_tree`. 

The default `manifest.js` should looks like this:
> //= link_tree ../images
//= link_directory ../javascripts .js
//= link_directory ../stylesheets .css

This is meant to include the contents of all files found in the `./app/assets/images` directory or any subdirectories as well as any file recognized as JS directly at `./app/assets/javascripts` or as CSS directly at `./app/assets/stylesheets` (both not including subdirectories).

-------------------------
Question
1. Why `//= link_directory ../javascripts .js` is not in this file?
2. Why `both not including subdirectories` in `./app/assets/javascripts` and `./app/assets/stylesheets`?

Reply:
Not bad. I wanted to ask that Q to you.

1. Because you might not want to include all your js files. Responsibility fall onto the dev to decide
2. This one, I dowan tell you. Go read up and understand about how rails create the final html,css,js & any other assets to serve to the client side then tell me why

Reply:
2. There might be some assets that are used infrequently, so it saves up the time for HTTP request when the subdirectory is not included. If later, one needs the subdirectory, they can use `link`.